### DISCUSSION TITLE
Usage in combinación with Revise.jl

### DISCUSSION THREAD

**Discussion by martinmestre** [rodluger/showyourwork#469] [General]
Hi all!
I would like to know if there is a way to rerun a script that uses a local package that has been modified. In other words, is there a way that showyourwork communicates with Revise.jl ?
Thanks !

**Comment by HealthyPear** [comment]
I am not an expert of Julia, so here is my interpretation:

- if you can add `using Revise` to your Julia script then everything should work, because any script is treated as an input file (i.e. if you modify it, the rule will be run again by Snakemake)
- if you cannot tell your script to use this package, the next idea might be that of [adding Revise to your Julia startup](https://timholy.github.io/Revise.jl/stable/config/#Using-Revise-by-default) then theoretically Snakemake will not care

**Comment by martinmestre** [comment]
Thanks! I will try both.  In the meanwhile I thought in a third option: adding the source file of my used packages (the ones I am normally tracking with Revise) to the snakemake pipeline. For example, if it is possible to add a whole directory,  I add to the pipeline:
```
input:
        "Manifest.toml"
        "~/.julia/dev/MyPackage/"
```
Is it correct?

**Comment by HealthyPear** [comment]
No it is not possible ti specify an entire directory as input, only output

if they are not many files it would be better to list them explicitly

**Comment by martinmestre** [comment]
Thanks a lot!
